ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -0,0 +1,707 @@
+      ledger =>
+        val n = 10000
+        val veryLongList = Primitive.List(List.iterate(0L, n)(_ + 1): _*)
+        for {
+          party <- ledger.allocateParty()
+          template = ParameterShowcase(
+            party,
+            42L,
+            BigDecimal("47.0000000000"),
+            "some text",
+            true,
+            Primitive.Timestamp.MIN,
+            NestedOptionalInteger(OptionalInteger.SomeInteger(-1L)),
+            veryLongList,
+            Primitive.Optional("some optional text")
+          )
+          create <- ledger.submitAndWaitRequest(party, template.create.command)
+          transaction <- ledger.submitAndWaitForTransaction(create)
+        } yield {
+          val contract = assertSingleton("VeryLongList", createdEvents(transaction))
+          assertEquals("VeryLongList", contract.getCreateArguments, template.arguments)
+        }
+    }
+
+  private[this] val notArchiveNonConsuming =
+    LedgerTest(
+      "TXNotArchiveNonConsuming",
+      "Expressing a non-consuming choice on a contract should not result in its archival") {
+      ledger =>
+        for {
+          Vector(receiver, giver) <- ledger.allocateParties(2)
+          agreementFactory <- ledger.create(giver, AgreementFactory(receiver, giver))
+          _ <- ledger.exercise(receiver, agreementFactory.exerciseCreateAgreement)
+          transactions <- ledger.flatTransactions(receiver, giver)
+        } yield {
+          assert(
+            !transactions.exists(_.events.exists(_.event.isArchived)),
+            s"The transaction include an archival: ${transactions.flatMap(_.events).filter(_.event.isArchived)}"
+          )
+        }
+    }
+
+  private[this] val requireAuthorization =
+    LedgerTest("TXRequireAuthorization", "Require only authorization of chosen branching signatory") {
+      ledger =>
+        for {
+          Vector(alice, bob) <- ledger.allocateParties(2)
+          template = BranchingSignatories(true, alice, bob)
+          _ <- ledger.create(alice, template)
+          transactions <- ledger.flatTransactions(alice)
+        } yield {
+          assert(template.arguments == transactions.head.events.head.getCreated.getCreateArguments)
+        }
+    }
+
+  private[this] val notDiscloseCreateToNonSignatory =
+    LedgerTest(
+      "TXNotDiscloseCreateToNonSignatory",
+      "Not disclose create to non-chosen branching signatory") { ledger =>
+      for {
+        Vector(alice, bob) <- ledger.allocateParties(2)
+        template = BranchingSignatories(false, alice, bob)
+        create <- ledger.submitAndWaitRequest(bob, template.create.command)
+        transaction <- ledger.submitAndWaitForTransaction(create)
+        transactions <- ledger.flatTransactions(alice)
+      } yield {
+        assert(transactions.find(_.transactionId != transaction.transactionId).isEmpty)
+      }
+    }
+
+  private[this] val discloseCreateToSignatory =
```

Comment:
This test is only partially useful. Could you put a comment that this should actually test that a party that is a signatory but not the submitter also receives the create event?

Reply:
Even better, I completed the test following your indications. Thanks for the help offline. :slightly_smiling_face: 

Addressed by https://github.com/digital-asset/daml/pull/2713/commits/3be1b233f0c85d4c46dfb48c245699fbe1ca4111